src/ui/ui_settings/create_or_upgrade_saved_config/__tests__/create_or_upgrade_saved_config.js

```diff
@@ -0,0 +1,120 @@
+import sinon from 'sinon';
+import Chance from 'chance';
+
+import * as getUpgradeableConfigNS from '../get_upgradeable_config';
+import { createOrUpgradeSavedConfig } from '../create_or_upgrade_saved_config';
+
+const chance = new Chance();
+
+describe('uiSettings/createOrUpgradeSavedConfig', function () {
+  const sandbox = sinon.sandbox.create();
+  afterEach(() => sandbox.restore());
+
+  const version = '4.0.1';
+  const prevVersion = '4.0.0';
+  const buildNum = chance.integer({ min: 1000, max: 5000 });
+
+  function setup() {
+    const log = sinon.stub();
+    const getUpgradeableConfig = sandbox.stub(getUpgradeableConfigNS, 'getUpgradeableConfig');
+    const savedObjectsClient = {
+      create: sinon.spy(async (type, attributes, options = {}) => ({
+        type,
+        id: options.id,
+        version: 1,
+      }))
+    };
+
+    async function run() {
+      const resp = await createOrUpgradeSavedConfig({
+        savedObjectsClient,
+        version,
+        buildNum,
+        log,
+      });
+
+      if (getUpgradeableConfig.callCount) {
```

Comment:
__nit:__ hmm, `getUpgradeableConfig` seems to be always called and is called only once or ...?

Reply:
At this point, that seems to be the case